base/iobuffer.jl

```diff
@@ -283,27 +410,41 @@ read(from::GenericIOBuffer, ::Type{Ptr{T}}) where {T} = convert(Ptr{T}, read(fro
 isreadable(io::GenericIOBuffer) = io.readable
 iswritable(io::GenericIOBuffer) = io.writable
 
+# Number of bytes that can be read from the buffer, if you seek to the start first.
 filesize(io::GenericIOBuffer) = (io.seekable ? io.size - io.offset : bytesavailable(io))
+
+# Number of bytes that can be read from the buffer.
 bytesavailable(io::GenericIOBuffer) = io.size - io.ptr + 1
+
+# Position is zero-indexed, but ptr is one-indexed, hence the -1
+# TODO: Document that position for an unseekable stream is invalid, or
+# make it error
 position(io::GenericIOBuffer) = io.ptr - io.offset - 1
```

Comment:
My vote here is for an error

Reply:
`mark` returns the "marked position" of the stream, so erroring will also make `mark` error.